File: string/1576_replace_all_?_s_to_avoid_consecutive_repeating_characters/1576_replace_all_?_s_to_avoid_consecutive_repeating_characters.py
import string


class Solution:
    def modifyString(self, s: str) -> str:
        l = list(s)
        alphabet = list(string.ascii_lowercase)
        for i, _ in enumerate(l):
            if l[i] == "?":
                for j in alphabet:
                    if i == len(l) - 1:
                        if l[i-1] != j:
                            l[i] = j
                            break
                    elif l[i-1] != j and l[i+1] != j:
                        l[i] = j
                        break
        return ''.join(l)


s = "?zs"
res = Solution().modifyString(s)
print(res)

File: string/1576_replace_all_?_s_to_avoid_consecutive_repeating_characters/1576_replace_all_?_s_to_avoid_consecutive_repeating_characters.go
package main

import "fmt"

func modifyString(s string) string {
	chars := make([]byte, len(s))
	for i := 0; i < len(chars); i++ {
		if s[i] != '?' {
			chars[i] = s[i]
			continue
		}
		for j := byte('a'); j <= 'z'; j++ {
			if (i == 0 || j != chars[i-1]) && (i == len(chars)-1 || j != s[i+1]) {
				chars[i] = j
				break
			}
		}
	}
	return string(chars)
}

func main() {
	s := "?zs"
	res := modifyString(s)
	fmt.Println(res)
}
